src/defineFunction.ts

```diff
@@ -1,51 +1,43 @@
-// @flow
-import type Parser from "./Parser";
-import type {ParseNode, AnyParseNode, NodeType, UnsupportedCmdParseNode}
-    from "./parseNode";
-import type Options from "./Options";
-import type {ArgType, BreakToken} from "./types";
-import type {HtmlDomNode} from "./domTree";
-import type {Token} from "./Token";
-import type {MathDomNode} from "./mathMLTree";
+import Parser from "./Parser";
+import { ParseNode, AnyParseNode, NodeType, UnsupportedCmdParseNode } from "./parseNode";
+import Options from "./Options";
+import { ArgType, BreakToken } from "./types";
+import { HtmlDomNode } from "./domTree";
+import { Token } from "./Token";
+import { MathDomNode } from "./mathMLTree";
 
 /** Context provided to function handlers for error messages. */
-export type FunctionContext = {|
+export type FunctionContext = {
     funcName: string,
     parser: Parser,
     token?: Token,
-    breakOnTokenText?: BreakToken,
-|};
+    breakOnTokenText?: BreakToken
+};
 
-export type FunctionHandler<NODETYPE: NodeType> = (
+export type FunctionHandler<NODETYPE extends NodeType> = ((
     context: FunctionContext,
     args: AnyParseNode[],
-    optArgs: (?AnyParseNode)[],
-) => UnsupportedCmdParseNode | ParseNode<NODETYPE>;
+    optArgs: AnyParseNode | null[]
+) => UnsupportedCmdParseNode | ParseNode<NODETYPE>);
 // Note: reverse the order of the return type union will cause a flow error.
 // See https://github.com/facebook/flow/issues/3663.
 
-export type HtmlBuilder<NODETYPE> = (ParseNode<NODETYPE>, Options) => HtmlDomNode;
-export type MathMLBuilder<NODETYPE> = (
-    group: ParseNode<NODETYPE>,
-    options: Options,
-) => MathDomNode;
+export type HtmlBuilder<NODETYPE> = ((parseNode: ParseNode<NODETYPE>, options: Options) => HtmlDomNode);
+export type MathMLBuilder<NODETYPE> = ((group: ParseNode<NODETYPE>, options: Options) => MathDomNode);
 
 // More general version of `HtmlBuilder` for nodes (e.g. \sum, accent types)
 // whose presence impacts super/subscripting. In this case, ParseNode<"supsub">
 // delegates its HTML building to the HtmlBuilder corresponding to these nodes.
-export type HtmlBuilderSupSub<NODETYPE> =
-    (ParseNode<"supsub"> | ParseNode<NODETYPE>, Options) => HtmlDomNode;
+export type HtmlBuilderSupSub<NODETYPE> = ((arg0: ParseNode<"supsub"> | ParseNode<NODETYPE>, options: Options) => HtmlDomNode);
 
 export type FunctionPropSpec = {
     // The number of arguments the function takes.
     numArgs: number,
-
```

Comment:
I personally like an extra space between before comment blocks, but if this is how prettier wants to do things I'll live with it.